[Mailer] Delivery randomly fails with Google SMTP Relay

Hello there,

## TL;DR:
Using _Google SMTP Relay_ with the new _Mailer Component_ works in a random manner.

-----

### Background
I'm working on a new project for which I chose the new _Mailer component_ over _Swift Mailer_ with the same configuration of what I'm used to (i.e. using [Google SMTP Relay](https://support.google.com/a/answer/2956491)).

This service accessible from Google Apps allows using Google's SMTP servers **without authentication**, on an IP whitelist basis. 

The Google SMTP Relay DSN is the following one, and doesn't require any credentials:

```bash
smtp://smtp-relay.gmail.com:587?encryption=&auth_mode=
```

### Issue
I've been using it with SwiftMailer for years without any issue, but when using _Symfony Mailer_, it is _likely_ to result in the following exception:

> {"exception":"[object] (Symfony\Component\Mailer\Exception\TransportException(code: 0): Expected response code "250" but got an empty response. at ~/builds/prod/releases/4/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php:236)","command":"app:test","message":"Expected response code "250" but got an empty response."}

The _likely_ is the problem, because that exception doesn't occur every time.

When adding a `dump($response);` after this line: 
https://github.com/symfony/symfony/blob/aa4385dc125213342c8292c9060693fa51cbad4d/src/Symfony/Component/Mailer/Transport/Smtp/SmtpTransport.php#L142 

There's a 3/5 chance the dumped response will be 

```bash
"421 4.7.0 Try again later, closing connection. (EHLO) x12sm193387wrq.21 - gsmtp\r\n"
""
````

over

```bash
"""
250-smtp-relay.gmail.com at your service, [92.154.***.***]\r\n
250-SIZE 157286400\r\n
250-8BITMIME\r\n
250-STARTTLS\r\n
250-ENHANCEDSTATUSCODES\r\n
250-PIPELINING\r\n
250-CHUNKING\r\n
250 SMTPUTF8\r\n
"""
"250 2.1.0 OK f50sm190172eda.33 - gsmtp\r\n"
"250 2.1.5 OK f50sm190172eda.33 - gsmtp\r\n"
"354  Go ahead f50sm190172eda.33 - gsmtp\r\n"
"250 2.0.0 OK  1561367145 f50sm190172eda.33 - gsmtp\r\n"
```

You'll probably tell me the problem comes from Google, since it's a `421` response code from their servers. The problem is that the issue just _never_ occurs with  _SwiftMailer_. Does the new component does some more things (like some sort of handshaking or whatever - I have no knowledge of how this works) Google SMTP servers can't interpret properly?

### How to reproduce
When I run the following command several times:

```bash
# .env
###> symfony/mailer ###
MAILER_DSN=smtp://smtp-relay.gmail.com:587?encryption=&auth_mode=
###< symfony/mailer ###
###> symfony/swiftmailer-bundle ###
MAILER_URL=smtp://smtp-relay.gmail.com:587?encryption=&auth_mode=
###< symfony/swiftmailer-bundle ###
```

```php
namespace App\Command;

use Symfony\Component\Console\Command\Command;
use Symfony\Component\Console\Input\InputInterface;
use Symfony\Component\Console\Output\OutputInterface;
use Symfony\Component\Console\Style\SymfonyStyle;
use Symfony\Component\Mailer\Exception\TransportExceptionInterface;
use Symfony\Component\Mailer\MailerInterface as SymfonyMailer;
use Symfony\Component\Mime\Email as SymfonyMailerEmail;
use Symfony\Component\Mime\Part\TextPart;

class TestCommand extends Command
{
    protected static $defaultName = 'app:test';

    /**
     * @var SymfonyMailer
     */
    private $symfonyMailer;
    
    /**
     * @var \Swift_Mailer
     */
    private $swiftMailer;

    public function __construct(SymfonyMailer $symfonyMailer, \Swift_Mailer $swiftMailer)
    {
        parent::__construct();
        $this->symfonyMailer = $symfonyMailer;
        $this->swiftMailer = $swiftMailer;
    }

    protected function execute(InputInterface $input, OutputInterface $output)
    {

        $io = new SymfonyStyle($input, $output);

        try {
            $io->comment('Sending through Symfony Mailer...');
            $message = new SymfonyMailerEmail();
            $message->setBody(new TextPart(\date(('Y-m-d H:i:s'))));
            $message->subject('Symfony Mailer');
            $message->addFrom('helloworld@mycustomdomain.com');
            $message->addTo('me@mycustomdomain.com');
            $this->symfonyMailer->send($message);
            $io->success('Done.');
        } catch (TransportExceptionInterface $e) {
            $io->error($e->getMessage());
        }

        try {
            $io->comment('Sending throught Swift Mailer...');
            $message = new \Swift_Message('Swift Mailer', \date(('Y-m-d H:i:s')));
            $message->addFrom('helloworld@mycustomdomain.com');
            $message->addTo('me@mycustomdomain.com');
            $this->swiftMailer->send($message);
            $io->success('Done.');
        } catch (\Exception $e) {
            $io->error($e->getMessage());
        }
    }
}
```

SwiftMailer _always_ succeeds, while the Mailer component succeeds _sometimes_.

Thank you,
Ben

Hello everyone, I'm back from vacations :-)

Good news, I just created a sample project with `symfony/mailer:4.3.4` and `symfony/google-mailer:4.3.4` and ran my example command: the messages get delivered! :rocket: 

Just 1 small issue remaining: when trying to send a message from a non-whitelisted IP address, here's the Swiftmailer console output (caught exception and $io->error(), cf my sample code at the top of the issue):

```
[ERROR] Expected response code "250" but got code "550", with message "550-5.7.1 Invalid credentials for relay         
                                                                        
                               n your G Suite SMTP Relay service doesn't match domain of
                               this email is being sent from. If you are trying to relay
                                 main that isn't registered under your G Suite account
                                     ope-from, you must configure your mail server
                                 SMTP AUTH to identify the sending domain or to present
                                      names in the HELO or EHLO command. For more
                                                                           
         550 5.7.1  https://support.google.com/a/answer/6140680#invalidcred i1sm130156edf.43 - gsmtp". 
```

And here's the output for Symfony Mailer:
```
PHP Fatal error:  Uncaught ErrorException: Notice: fwrite(): send of 6 bytes failed with errno=32 Broken pipe in /home/ben/PhpstormProjects/test-mailer/vendor/symfony/mailer/Transport/Smtp/Stream/AbstractStream.php:38
Stack trace:
#0 /home/ben/PhpstormProjects/test-mailer/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(145): Symfony\Component\Mailer\Transport\Smtp\Stream\AbstractStream->write('QUIT\r\n')
#1 /home/ben/PhpstormProjects/test-mailer/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(208): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->executeCommand('QUIT\r\n', Array)
#2 /home/ben/PhpstormProjects/test-mailer/vendor/symfony/mailer/Transport/Smtp/SmtpTransport.php(288): Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->stop()
#3 [internal function]: Symfony\Component\Mailer\Transport\Smtp\SmtpTransport->__destruct()
#4 {main}
  thrown in /home/ben/PhpstormProjects/test-mailer/vendor/symfony/mailer/Transport/Smtp/Stream/AbstractStream.php on line 38
```

--- a/src/Symfony/Component/Mailer/Transport/Smtp/Stream/AbstractStream.php
+++ b/src/Symfony/Component/Mailer/Transport/Smtp/Stream/AbstractStream.php
@@ -35,7 +35,7 @@ abstract class AbstractStream
         $bytesToWrite = \strlen($bytes);
         $totalBytesWritten = 0;
         while ($totalBytesWritten < $bytesToWrite) {
-            $bytesWritten = fwrite($this->in, substr($bytes, $totalBytesWritten));
+            $bytesWritten = @fwrite($this->in, substr($bytes, $totalBytesWritten));
             if (false === $bytesWritten || 0 === $bytesWritten) {
                 throw new TransportException('Unable to write bytes on the wire.');
             }

